lib/active_model/serializer/collection_serializer.rb

```diff
@@ -23,8 +23,11 @@ def initialize(resources, options = {})
       end
 
       def json_key
-        key = root || serializers.first.try(:json_key) || object.try(:name).try(:underscore)
-        key.try(:pluralize)
+        root || begin
+          singular_key = serializers.first.try(:json_key) ||
+            object.try(:name).try(:underscore)
+          singular_key.try(:pluralize)
+        end
```

Comment:
what a mess this code is.  maybe have this be something like `root || derived_root` so it's a little easier to see what's going on?  I initially missed that the fundamental change was no longer modifying `root` when present